### DISCUSSION TITLE
npm error on driver/plugin installation

### DISCUSSION THREAD

**Discussion by RSM-SBA** [appium/appium#17109] [Q&A] [answered]
can someone help me with this npm error which occurs when i install appium@beta40 and drivers /plugins

> npx appium plugin uninstall images || true && npx appium driver uninstall uiautomator2 || true && npx appium driver install uiautomator2 || true && npx appium plugin install images || true

Error: Can't uninstall plugin 'images'; it is not installed
Error: Can't uninstall driver 'uiautomator2'; it is not installed

- Installing 'uiautomator2' using NPM install spec 'appium-uiautomator2-driver'
✖ Installing 'uiautomator2' using NPM install spec 'appium-uiautomator2-driver'
Error: Encountered an error when installing package: npm command 'install --save-dev appium-uiautomator2-driver --json' failed with code 1.

STDERR:
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: appium_installation@1.0.0
npm ERR! Found: appium@2.0.0-beta.30
npm ERR! node_modules/appium
npm ERR!   appium@"2.0.0-beta.30" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer appium@"^2.0.0-beta.40" from appium-uiautomator2-driver@2.4.0
npm ERR! node_modules/appium-uiautomator2-driver
npm ERR!   dev appium-uiautomator2-driver@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/jenkins/.npm/eresolve-report.txt for a full report.
{
  "error": {
    "code": "ERESOLVE",
    "summary": "unable to resolve dependency tree",
    "detail": "\nWhile resolving: appium_installation@1.0.0\nFound: appium@2.0.0-beta.30\nnode_modules/appium\n  appium@\"2.0.0-beta.30\" from the root project\n\nCould not resolve dependency:\npeer appium@\"^2.0.0-beta.40\" from appium-uiautomator2-driver@2.4.0\nnode_modules/appium-uiautomator2-driver\n  dev appium-uiautomator2-driver@\"*\" from the root project\n\nFix the upstream dependency conflict, or retry\nthis command with --force, or --legacy-peer-deps\nto accept an incorrect (and potentially broken) dependency resolution.\n\nSee /Users/jenkins/.npm/eresolve-report.txt for a full report."
  }
}

**Comment by KazuCocoa** [comment, marked as answer]
what's about upgrading the appium beta from 30 to 40?

**Reply by RSM-SBA** [reply]
good hint, i had a typo in my repo branch naming ;)
i haven't used the package.json i thought :D

**Comment by Karthikeyan021298** [comment]
@KazuCocoa , I am facing the  Error, when trying to download Mac2 Driver  and XCUITest Driver- Appium version- - 2.0.0-beta.46,  Node version - v18.11.0, npm version -- 8.19.2

can anyone please help on this issue, I am blocked


✖ Installing 'mac2' using NPM install spec 'appium-mac2-driver'
[31mError: ✖ Encountered an error when installing package: npm command 'install --save-dev appium-mac2-driver --json' failed with code 1.

STDOUT:


STDERR:
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: undefined@undefined
npm ERR! Found: appium@1.22.3
npm ERR! node_modules/appium
npm ERR!   appium@"^1.21.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer appium@"^2.0.0-beta.44" from appium-mac2-driver@1.4.4
npm ERR! node_modules/appium-mac2-driver
npm ERR!   dev appium-mac2-driver@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/macbookair1/.npm/eresolve-report.txt for a full report.
{
  "error": {
    "code": "ERESOLVE",
    "summary": "unable to resolve dependency tree",
    "detail": "\nWhile resolving: undefined@undefined\nFound: appium@1.22.3\nnode_modules/appium\n  appium@\"^1.21.0\" from the root project\n\nCould not resolve dependency:\npeer appium@\"^2.0.0-beta.44\" from appium-mac2-driver@1.4.4\nnode_modules/appium-mac2-driver\n  dev appium-mac2-driver@\"*\" from the root project\n\nFix the upstream dependency conflict, or retry\nthis command with --force, or --legacy-peer-deps\nto accept an incorrect (and potentially broken) dependency resolution.\n\nSee /Users/macbookair1/.npm/eresolve-report.txt for a full report."
  }
}

Thanks in Advance,
Karthikeyan

**Reply by KazuCocoa** [reply]
I haven't met the error, but it seems strange.
Did your environment have `appium@1.22.3` or... `appium@1.21.0`, or something dependencies related to them?

**Reply by Karthikeyan021298** [reply]
I was using appium@1.22.3 in my machince and I unistalled the appium@1.22.3 using "npm uninstall -g appium" and upgraded to the appium 2 beta version, after that I was facing in the installation of Mac2 Driver and XCUITest Drivers

**Reply by KazuCocoa** [reply]
Where was the `node_modules` you installed appium v1?

**Reply by Karthikeyan021298** [reply]
@KazuCocoa  "/opt/homebrew/lib/node_modules/appium"

**Reply by KazuCocoa** [reply]
https://github.com/appium/appium/issues/17683 similar issue..?

I haven't used node via homebrew, but I wonder if you need to provide `sudo` to uninstall/install node packages via `npm` installed via homebrew.

**Comment by felixtruong-vn** [comment]
You can resolve problem following criterias: 
Navigate to users/userName directory and delete the following folder and files

/node_modules/
package.json
package.lock.json
These above files refer to earlier appium installation.

then try to run command to install appium drivers, you should be good now!

**Reply by Karthikeyan021298** [reply]
Yes I did, It's working, now I can able to download the Drivers @giangthao .

Thank you @KazuCocoa  & @giangthao,
Regards,
@Karthikeyan021298

**Reply by Hemanth-Rezi** [reply]
> You can resolve problem following criterias: Navigate to users/userName directory and delete the following folder and files
> 
> /node_modules/ package.json package.lock.json These above files refer to earlier appium installation.
> 
> then try to run command to install appium drivers, you should be good now!

Yes this works for me as well. Thanks @giangthao

**Reply by jlipps** [reply]
@boneskull i wonder if we should search up the directory tree from `APPIUM_HOME` to see if there are any `package.json`s above it, which could cause issues, and at least emit a warning? at some point i also ended up with a `package.json` in my homedir unintentionally, and it caused problems until i removed it.
